tests/network/link_state.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2025 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = SIGDescribe("interface state up/down", func() {
+
+	Context("VM with one link up and  one link down", func() {
+		const (
+			primaryLogicalNetName    = "default"
+			secondary2LogicalNetName = "bridge2"
+			nadName                  = "bridge-nad"
+		)
+
+		var (
+			vm            *v1.VirtualMachine
+			mac1, mac2    net.HardwareAddr
+			testNamespace string
+		)
+
+		BeforeEach(func() {
+			testNamespace = testsuite.GetTestNamespace(nil)
+
+			var err error
+			_, err = libnet.CreateNetAttachDef(context.Background(), testNamespace,
+				libnet.NewBridgeNetAttachDef(nadName, "br02"))
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		BeforeEach(func() {
+
+			var err error
+			mac1, err = libnet.GenerateRandomMac()
+			Expect(err).NotTo(HaveOccurred())
+			mac2, err = libnet.GenerateRandomMac()
+			Expect(err).NotTo(HaveOccurred())
+
+			vmi := libvmifact.NewFedora(
```

Comment:
Please consider the following alternative:

```go
			primaryIface := libvmi.InterfaceDeviceWithMasqueradeBinding()
			primaryIface.MacAddress= mac1.String()
			primaryIface.State = v1.InterfaceStateLinkUp
			
			secondaryIface := libvmi.InterfaceDeviceWithBridgeBinding(secondary2LogicalNetName)
			secondaryIface.MacAddress= mac2.String()
			secondaryIface.State = v1.InterfaceStateLinkDown
			
			vmi := libvmifact.NewFedora(
				libvmi.WithInterface(primaryIface),
				libvmi.WithNetwork(v1.DefaultPodNetwork()),
				libvmi.WithInterface(secondaryIface),
				libvmi.WithNetwork(libvmi.MultusNetwork(secondary2LogicalNetName, nadName)),
			)
```

Reply:
Could you please link the comment?

Reply:
My suggestion is a nice to have.
I personally think it is more readable, but I will not block on it.